schema.go

```diff
@@ -41,156 +54,214 @@ func scheYAML(rootSchema *jsonschema.Schema, cfg *Config) (*yaml.Node, error) {
+		keyNode := &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Value: propertyName,
+		}
 
-			continue
+		// add a HeadComment to the schema if a node is found which has a description or examples
+		schemaWithDescription, hasDescription := coalesce(schemas, withDescription)
+		schemaWithExamples, hasExamples := coalesce(schemas, withExamples)
+		switch {
+		case hasDescription && hasExamples:
+			keyNode.HeadComment = formatHeadComment(*schemaWithDescription.Description, schemaWithExamples.Examples, cfg.LineLength)
+		case hasDescription:
+			keyNode.HeadComment = formatHeadComment(*schemaWithDescription.Description, []any{}, cfg.LineLength)
+		case hasExamples:
+			keyNode.HeadComment = formatHeadComment("", schemaWithExamples.Examples, cfg.LineLength)
 		}
 
-		// The property value node
-		valueNode, err := scheYAML(property, cfg.forProperty(propertyName))
+		// else recursively determine the nodeValue using scheYAML
+		valueNode, err := scheYAML(rootschema, cfg.forProperty(propertyName, patterns))
 		if err != nil {
 			return nil, fmt.Errorf("failed to scheyaml %q: %w", propertyName, err)
 		}
 
-		if valueNode.Content == nil && valueNode.Kind == yaml.MappingNode {
+		// in case only
+		if len(valueNode.Content) == 0 && valueNode.Kind == yaml.MappingNode {
 			valueNode.Value = "{}"
 		}
 
-		if patternNodes, ok := patternProperties[propertyName]; ok {
-			valueNode.Content = append(valueNode.Content, patternNodes...)
-		}
-
-		result = append(result, valueNode)
+		result = append(result, keyNode, valueNode)
 	}
 
 	return result, nil
 }
 
-// determinePatternProperties's purpose is to generate additional nodes for properties that match
-// defined patternProperties in the schema
-func determinePatternProperties(schema *jsonschema.Schema, cfg *Config) (map[string][]*yaml.Node, error) {
-	result := make(map[string][]*yaml.Node)
-
-	if schema.Properties == nil || schema.PatternProperties == nil {
-		return result, nil
+// resolve returns a new slice in which schemas that are references are replaced with the resolved reference
+func resolve(schemas []*jsonschema.Schema) []*jsonschema.Schema {
+	if len(schemas) == 0 {
+		return schemas
 	}
 
-	properties := maps.Keys(*schema.Properties)
-
-	for regex, patternProperty := range *schema.PatternProperties {
-		parsedRegex, err := regexp.Compile(regex)
-		if err != nil {
-			return nil, fmt.Errorf("failed to parse %q as a regex: %w", regex, err)
+	res := make([]*jsonschema.Schema, len(schemas))
+	for i, s := range schemas {
+		if s != nil && s.Ref != "" {
+			res[i] = s.ResolvedRef
+		} else {
+			res[i] = s
 		}
+	}
 
-		for _, property := range properties {
-			if !parsedRegex.MatchString(property) {
-				continue
-			}
+	return res
+}
 
-			result[property], err = scheYAMLObject(patternProperty, cfg)
-			if err != nil {
-				return nil, fmt.Errorf("failed to scheyaml %q: %w", regex, err)
-			}
+// patternProperties returns matching pattern properties sorted in alphabetical order for some property name
```

Comment:
Could you rename this to `matchingPatternProperties` or something for clarity? That that is what it does. `patternProperties` on its own is already a commonly used variable name